### PR TITLE
Modify rule S2225: Add VB.NET

### PR DESCRIPTION
**Pull request by Corniel**
See https://github.com/SonarSource/sonar-dotnet/pull/5741

### CODE REVIEW

**Comment by Corniel** [comment]
@arseniy-sonar The tests failed due to insufficient rights. Why is that? I assume it was not triggered by my (I obviously do not have the rights)?!

**Comment by arseniy-sonar** [comment]
Hi @Corniel 
Can you, please, retrigger the analysis?

**Comment by Corniel** [comment]
> Hi @Corniel Can you, please, retrigger the analysis (by committing an insignificant change)?

A re-base will do, I suppose. :)

**Comment by arseniy-sonar** [comment]
> A re-base will do, I suppose. :)
Yep. Thank you.
The repository was misconfigured (hiding secrets from CI triggered by external collaborators). Now it does not do that, but still requires approval from a SonarSourcer to run the CI for security purposes.

**Comment by Corniel** [comment]
@pavel-mikula-sonarsource I assume you're the candidate to review this?

**Comment by pavel-mikula-sonarsource** [comment]
@Corniel Yes, thanks. I've assigned myself to all VB.NET related PRs.

**Comment by Corniel** [comment]
@pavel-mikula-sonarsource 
> ERROR: Connection error HTTPSConnectionPool(host='dl.packetstormsecurity.net', port=443): Max retries exceeded with url: /papers/attack/HTML5AttackVectors_RafayBaloch_UPDATED.pdf (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)')))
Welcome into certificate hell.

**Comment by Corniel** [comment]
@pavel-mikula-sonarsource  Might be worth picking this up?

File: rules/S2225/vbnet/rule.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Calling ``++ToString()++`` on an object should always return a string. Returning ``++Nothing++`` instead contravenes the method's implicit contract.
+
+== Noncompliant Code Example
+
+[source,csharp]
```
Comment (pavel-mikula-sonarsource) [inline rules/S2225/vbnet/rule.adoc:5]:
```suggestion
[source,vbnet]
```

File: rules/S2225/vbnet/rule.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
+Calling ``++ToString()++`` on an object should always return a string. Returning ``++Nothing++`` instead contravenes the method's implicit contract.
```
Comment (pavel-mikula-sonarsource) [inline rules/S2225/vbnet/rule.adoc:1]: The original encoding is some noise from importing data
```suggestion
Calling `ToString()` on an object should always return a string. Returning `Nothing` instead contravenes the method's implicit contract.
```

File: rules/S2225/vbnet/rule.adoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+== Noncompliant Code Example
+
+[source,csharp]
+----
+public override string ToString ()
+{
```
Comment (pavel-mikula-sonarsource) [inline rules/S2225/vbnet/rule.adoc:8]: Copy/paste issue, should not be here
```suggestion
```

File: rules/S2225/vbnet/rule.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+Public Overrides Function ToString() As String
+    Return Nothing
+End Function
+}
```
Comment (pavel-mikula-sonarsource) [inline rules/S2225/vbnet/rule.adoc:12]: Same
```suggestion
```

File: rules/S2225/vbnet/rule.adoc [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+----
+
+== Compliant Solution
+
+[source,csharp]
```
Comment (pavel-mikula-sonarsource) [inline rules/S2225/vbnet/rule.adoc:17]:
```suggestion
[source,vbnet]
```

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Some copy/paste issues

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

File: rules/S2225/vbnet/rule.adoc [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+End Function
+----
+
+== Compliant Solution
+
+[source,vbnet]
+----
+Public Overrides Function ToString() As String
+    Return ""
+End Functio
```
Comment (pavel-mikula-sonarsource) [inline rules/S2225/vbnet/rule.adoc:18]: Typo
```suggestion
End Function
```

**Pull request merged**